Refactor for v3

MegaAdragon opened the pull request:
Refactor the module for companion v3

-> try to resolve #12

MegaAdragon commented on companion/manifest.json:
```diff
@@ -0,0 +1,31 @@
+{
+	"id": "phillips-hue",
```
Keep this type in order to overwrite the old module

Julusian replied:
you can change this if you add the old spelling to the `legacyIds` field further down

MegaAdragon commented on index.js:
```diff
@@ -1,497 +1,191 @@
+		this.log('debug', 'destroy')
+	}
+
+	async configUpdated(config) {
```
Somehow this is never called... Not sure why...

MegaAdragon commented:
@josephdadams do you want to review the changes?
Or can I just merge the branch with all the changes?

I tested the module and everything should work as before. Because I had to start from scratch and renamed a lot of stuff, I’m not sure what will happen when there is an existing v2 configuration…

Julusian commented on companion/manifest.json:
```diff
@@ -0,0 +1,31 @@
+	"runtime": {
+		"type": "node18",
+		"api": "nodejs-ipc",
+		"apiVersion": "1.0.0",
```
```suggestion
		"apiVersion": "0.0.0",
```

This is why `configUpdated` is never called

Julusian commented on package.json:
```diff
@@ -1,30 +1,21 @@
-	"shortname": "philips-hue",
-	"description": "Philips Hue module for Companion",
+	"name": "philips-hue",
+	"version": "0.2.0",
```
this should probably be bumped to 1.0.0 now

Julusian commented:
> I tested the module and everything should work as before. Because I had to start from scratch and renamed a lot of stuff, I’m not sure what will happen when there is an existing v2 configuration…

Yeah, it looks like every action has been renamed, so existing users will have all their actions break when they update companion to the version that includes this.

It would be a good idea to write an upgradescript to translate existing configs across. Something like https://github.com/bitfocus/companion-module-bmd-atem/blob/master/src/upgrades.ts#L131-L149, but perhaps with some updating of the options too

gauthierGrsse commented:
> Refactor the module for companion v3
> 
> -> try to resolve #12

Do you have some news for this bug fix for V3.0 ?

josephdadams commented:
I believe we are waiting on the PR author to add the upgrade scripts so that this doesn’t break the config for the hundreds of users

MegaAdragon commented:
> > I tested the module and everything should work as before. Because I had to start from scratch and renamed a lot of stuff, I’m not sure what will happen when there is an existing v2 configuration…
> 
> Yeah, it looks like every action has been renamed, so existing users will have all their actions break when they update companion to the version that includes this. For reference, there are at least a few hundred users of this module
> 
> It would be a good idea to write an upgradescript to translate existing configs across. Something like https://github.com/bitfocus/companion-module-bmd-atem/blob/master/src/upgrades.ts#L131-L149, but perhaps with some updating of the options too

@Julusian 
I looked at https://github.com/bitfocus/companion-module-base/wiki/Upgrade-scripts and tried to figure out an upgrade script, but I did not really come far... I found no possibility the debug the script properly.

Because I refactored + cleaned up the actions  completely, I would propose to just accept the broken actions for users that come from the old version of this module. I'm working on feedbacks and presets, so there is in actual benefit in manually restoring the button.

What do you think?

MegaAdragon commented:
Resolve #10 
and 
Resolve #16 (duplicate for #12)

MegaAdragon commented:
From my side, I'm ready to merge this branch. As mentioned, this will break configs from previous versions, but I completely overhauled the module...

SirAuran commented:
Hi are there any updates? Thanks a lot for the effort.
